### ISSUE TITLE
potential-bugs HasPlatformType does not work

### ISSUE THREAD

**Issue by kepocnhh** [arturbosch/detekt#2042]
```kotlin
private fun getPlatformType() = System.getProperty("propertyName")
```

## Expected Behavior
The rule will work and the detekt will report an error.

## Observed Behavior
Detekt does not report an error.

## Steps to Reproduce
I use my task:
```kotlin
task<Detekt>("verifyQuality") {
    source = files(subprojects.withPlugin("kotlin").srcDirs("main")).asFileTree
    config = files(
        "buildSrc/src/main/resources/detekt/config/potential_bugs.yml"
    )
    reports {
        html {
            enabled = true
            destination = File(analysisQualityHtmlPath)
        }
        xml.enabled = false
        txt.enabled = false
    }
}
```

File potential_bugs.yml contains:
```yaml
build:
  maxIssues: 0

potential-bugs:
  active: true
  HasPlatformType:
    active: true

processors:
  active: false
```

## Context
Detekt definitely works on other rule sets (for example, DuplicateCaseInWhenExpression). That is, the problem should not be in the source of the source code or the path to detekt configuration file. For some reason I can’t turn on this particular rule (Deprecation).
❗️ The code examples almost exactly copy the examples from the site.
- https://arturbosch.github.io/detekt/potential-bugs.html#noncompliant-code-4

## Environment
java 1.8.0_221
kotlin 1.3.50
detekt 1.1.1
Gradle 5.5.1
macOS Mojave Version 10.14.6 (18G87)

**Comment by arturbosch** [comment]
Same as #2038
